tests/core/string/test_string.h

```diff
@@ -1679,7 +1679,7 @@ TEST_CASE("[String] validate_node_name") {
 	CHECK(name_with_kana.validate_node_name() == U"Name with kana ゴドツ");
 
 	String name_with_invalid_chars = "Name with invalid characters :.@removed!";
-	CHECK(name_with_invalid_chars.validate_node_name() == "Name with invalid characters removed!");
+	CHECK(name_with_invalid_chars.validate_node_name() == "Name with invalid characters ___removed!");
```

Comment:
* This closes godotengine/godot-proposals#6225

I'm not sure which behavior is more desirable for users here: 1. (old) remove invalid characters, 2. replace each invalid character with `_`, or 3. replace the entire group of invalid characters with a single `_`.

* See also #67701 (it is not optimized).

Reply:
In this case, as this is done for performance reasons, the approach in this PR is probably the simplest.